### DISCUSSION TITLE
Data from AAPS w/o NS Client

### DISCUSSION THREAD

**Discussion by libxmike** [NightscoutFoundation/xDrip#2252] [Ideas]
The AAPS NS Client is a problem as it drains too much battery, sometimes over 1% per h, is there a way to avoid using NS Client to get the data to xDrip+ and upload to nightscout website?

**Comment by libxmike** [comment]
I get the AAPS data in the extra status line but not in the chart, and it is not uploaded to nightscout, only if AAPS NS Client is enabled for upload, then both, xDrip+ and NS Client will upload the data and I have double entry in nightscout.
![Screen Shot 2022-07-31 at 11 52 25 AM](https://user-images.githubusercontent.com/16604543/182020815-cc281370-0681-4bdb-a6d1-c2b0a1691338.png)

**Reply by jamorham** [reply]
I have a private fork of AAPS which implements this. Basically just emits broadcast from AAPS without implementing NS Client. This is really a change idea for AAPS side rather than xDrip afaik.

**Reply by libxmike** [reply]
Yeah, I totally agree with you, this was working some time ago in an old AAPS version.
I was already tinkering how to do it myself, but life and work project keeping me from this, I also opened issue on APPS git. 
Can you share your AAPS fork with that modification? I already have some customizations on my AAPS code.

**Reply by 4RK4N** [reply]
> I have a private fork of AAPS which implements this. Basically just emits broadcast from AAPS without implementing NS Client. This is really a change idea for AAPS side rather than xDrip afaik.

could you also share me that fork of aaps please? i tried to understand nsclient code but that didnt go so well :D
